assets/js/components/adminbar/AdminBarSessions.js

```diff
@@ -60,18 +58,21 @@ const AdminBarSessions = ( { className } ) => {
 		],
 		url,
 	};
+
 	const analyticsData = useSelect( ( select ) => select( MODULES_ANALYTICS ).getReport( reportArgs ) );
 	const hasFinishedResolution = useSelect( ( select ) => select( MODULES_ANALYTICS ).hasFinishedResolution( 'getReport', [ reportArgs ] ) );
 	const error = useSelect( ( select ) => select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [ reportArgs ] ) );
 
+	const reportZero = isZeroReport( analyticsData );
+	// Memoise the WidgetReportZero component to avoid Maximum update depth exceeded error.
+	const zeroDataComponent = useMemo( () => <WidgetReportZero moduleSlug="analytics" widgetSlug={ WIDGET_SLUG } />, [ reportZero ] );
+	if ( reportZero ) {
+		return zeroDataComponent;
+	}
```

Comment:
This should replace the `ReportZero` which is still in this component after the `error` condition below.